### PR TITLE
BIP143 example and clarify

### PR DESCRIPTION
**Pull request by jl2012**

### CODE REVIEW

**Comment by jl2012** [comment]
@sipa please review the scriptCode part.

I have doubled checked the example and verified the sigHash

File: bip-0143.mediawiki [hunk 1]
```diff
@@ -24,35 +24,41 @@ Deploying the aforementioned fixes in the original script system is not a simple
...
+     9. nLocktime of the transaction (4-byte little endian)
+    10. sighash type of the signature (4-byte little endian)
+
+The items 1, 4, 7, 9, 10 have the same meaning as the original algorithm. <ref name=wiki></ref>
+
+The item 5:
+*For P2WPKH witness program, the scriptCode is <code>0x1976a914{20-byte-pubkey-hash}88ac</code>.
+*For P2WSH witness program,
+**if the <code>witnessScript</code> does not contain any <code>OP_CODESEPERATOR</code>, the <code>scriptCode</code> is a <code>varInt</code> for the length of the <code>witnessScript</code>, followed by the <code>witnessScript</code>.
+**if the <code>witnessScript</code> contains any <code>OP_CODESEPERATOR</code>, the <code>scriptCode</code> is the evaluated script, with all <code>OP_CODESEPARATOR</code> and everything up to the last <code>OP_CODESEPARATOR</code> before the signature checking opcode being executed removed, and prepended by a <code>varInt</code> for the length of the trancated script.
```
Comment (sipa) [inline bip-0143.mediawiki]: The length is not prepended again.

**Comment by CodeShark** [comment]
ACK

**Comment by btcdrak** [comment]
ACK

**Comment by harrycheng8** [comment]
#921 #1118

**Review by LadyS8214 — APPROVED** [review]

**Pull request merged**